### ISSUE TITLE
Memory allocation problem

### ISSUE THREAD

**Issue by palles77** [julius-speech/julius#24]
I consitently seem to be getting problems with memory allocation on Windows. I tried to change value of MYBMALLOC_BLOCK_SIZE from 5000 to smaller or higher, to see how memory fragmentation is affecting the allocation, but for some reason I get the same memory block size problem with a message: 
Error: mymalloc: failed to allocate 3576400 bytes
It does not matter how long the input WAVE file is.
Below is my sample JCONF file:

-input file
-filelist Models\\test.dbl
-htkconf Models\\wav_config
-h Models\\PLPL-v4.2.am
-hlist Models\\PLPL-v4.2.phn
-d Models\\PLPL-v4.2.lm
-v Models\\PLPL-v4.2.dct
-b 2000
-lmp 12 -4
-lmp2 12 -4 
-walign 
-fallback1pass 
-multipath 
-iwsp
-norealtime
-iwcd1 max
-spmodel sp
-spsegment
-gprune none
-no_ccd
-sepnum 150
-b2 360 
-n 40 
-s 2000 
-m 8000 
-lookuprange 5 
-sb 80
-forcedict

**Comment by nitslp-ri** [comment]
I guess it is not a problem of memory fragment, maybe other memory leak.  Could you determine where it occurs by debugger?  For example, using GNU debugger:
```
% make distclean
% CFLAGS=-g ./configure  (or env CFLAGS=-g ./configure)
% make
% gdb ./julius/julius
% run [command line arguments]
```
When stopped by error, type `frame` to see from where the crashed mymalloc() was called.

Or could you show me the error log?

**Comment by palles77** [comment]
The bug is caused by Julius not being able to find a segment and it constantly allocates more and more memory, until it runs out. I played a lot with Julius over the years and one of the changes I made was to increase the value of MAXSPEECHLEN from 320000 to 32000000. However I managed to get around the bug by adding option -cutsilence, which forces the Julius to cut segments based on silence detection. This has improved not only decoding speed but also WER from around 11.5% to around 11%. I suppose for the time being I will close the bug, as there is a workaround available.
